sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -946,6 +958,25 @@ private <T> Flux<FeedResponse<T>> createQueryInternal(
                     }
                     return tFeedResponse;
                 });
+
+            CosmosEndToEndOperationLatencyPolicyConfig endToEndPolicyConfig =
+                options.getCosmosEndToEndOperationLatencyPolicyConfig() != null ?
+                options.getCosmosEndToEndOperationLatencyPolicyConfig() : this.cosmosEndToEndOperationLatencyPolicyConfig;
+
+            if (endToEndPolicyConfig != null && endToEndPolicyConfig.isEnabled()) {
+                return feedResponseFlux
+                    .timeout(endToEndPolicyConfig.getEndToEndOperationTimeout())
+                    .onErrorMap(throwable -> {
+                        if (throwable instanceof TimeoutException) {
```

Comment:
We should unwrap the throwable here, sometimes the error can be inside ReactorException or ReactiveException, please check at other places where we unwrap.